floop-data-export.py

```diff
@@ -38,7 +40,7 @@
 
     for convo in conversations:
         convo_entries = convo.reference.collection(
-            'Messages').order_by(u'Date_Submitted').limit(1).get()
+            'Messages').order_by(u'Date_Submitted').get()
 
         for entry in convo_entries:
             cList.append(entry.get("Text").strip())
```

Comment:
You should append an object _with_ the `Text` instead of just the `Text` itself, that way you can also insert a field that says who sent the message (a teacher or a student, no PII, uids). Something like this:
```ts
clist.append({
    Text: entry.get("Text").strip()),
    Sender: "Teacher" | "Student",
})
```

Note: this will affect how you're filtering out dupes.

Reply:
Okay Dan, I have appended dictionaries to the list instead of strings. Hope that was what you were referring too. Will create a pull request for the new code